### PR TITLE
feat: Add an utility to open or redirect for a deeplink

### PR DESCRIPTION
**Pull request by Crash--**
I don't know how to test this part since it's already a bit hacky by itself. 

This code is currently served by the stack, but we plan to use it also in the bar.

### CODE REVIEW

File: packages/cozy-device-helper/src/deeplink.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    failCb()
+    handler && handler.remove()
+  }, 500)
+
+  let iframe = document.getElementById('hiddenIframe')
+  if (!iframe) {
+    iframe = _createHiddenIframe(document.body, 'about:blank')
+  }
+
+  var handler = window.addEventListener('blur', onBlur)
```
Comment (y-lohse) [inline packages/cozy-device-helper/src/deeplink.js]: Why use `var` here?

File: packages/cozy-device-helper/src/deeplink.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  return iframe
+}
+
+const openUriWithHiddenFrame = (uri, failCb) => {
+  const timeout = setTimeout(function() {
+    failCb()
+    handler && handler.remove()
+  }, 500)
+
+  let iframe = document.getElementById('hiddenIframe')
```
Comment (y-lohse) [inline packages/cozy-device-helper/src/deeplink.js]: There's no risk if the iframe already exists? Wouldn't it be safer to create a new iframe every time?

File: packages/cozy-device-helper/src/deeplink.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  iframe.id = 'hiddenIframe'
+  iframe.style.display = 'none'
+  target.appendChild(iframe)
+  return iframe
+}
+
+const openUriWithHiddenFrame = (uri, failCb) => {
+  const timeout = setTimeout(function() {
+    failCb()
+    handler && handler.remove()
```
Comment (y-lohse) [inline packages/cozy-device-helper/src/deeplink.js]: I didn't know `handler.remove` could be a thing? [The documentation says `addEventListener` returns nothing](https://developer.mozilla.org/fr/docs/Web/API/EventTarget/addEventListener). What's the difference with `window.removeEventListener('blur', onBlur)`?

File: packages/cozy-device-helper/src/deeplink.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+const openUriWithMsLaunchUri = (uri, failCb) => {
+  navigator.msLaunchUri(uri, undefined, failCb)
+}
+
+const checkBrowser = () => {
+  const isOpera = !!window.opera || navigator.userAgent.indexOf(' OPR/') >= 0
+  const ua = navigator.userAgent.toLowerCase()
+  const isSafari =
+    (~ua.indexOf('safari') && !~ua.indexOf('chrome')) ||
```
Comment (y-lohse) [inline packages/cozy-device-helper/src/deeplink.js]: TIL https://www.joezimjs.com/javascript/great-mystery-of-the-tilde/ but uh… still a bit confused by this.

File: packages/cozy-device-helper/src/deeplink.js (hunk 3 above)
Comment (Crash--) [reply packages/cozy-device-helper/src/deeplink.js]: Old code, is old. Thank you. Fixed

File: packages/cozy-device-helper/src/deeplink.js (hunk 2 above)
Comment (Crash--) [reply packages/cozy-device-helper/src/deeplink.js]: Added a randomId

File: packages/cozy-device-helper/src/deeplink.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    .replace(/[^a-z]+/g, '')
+    .substr(0, 5)
+  window.addEventListener('blur', onBlur)
+  const timeout = setTimeout(function() {
+    failCb()
+    window.removeEventListener('blur', onBlur)
+  }, 500)
+
+  let iframe = document.getElementById(`hiddenIframe_${randomId}`)
+  if (!iframe) {
```
Comment (y-lohse) [inline packages/cozy-device-helper/src/deeplink.js]: I think we might as well remove the check entirely now that the id is random?

File: packages/cozy-device-helper/src/deeplink.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    iframe = _createHiddenIframe(document.body, 'about:blank', randomId)
+  }
+
+  function onBlur() {
+    clearTimeout(timeout)
+    window.removeEventListener('blur', onBlur)
+    iframe.parentElement.removeChild(iframe)
```
Comment (y-lohse) [inline packages/cozy-device-helper/src/deeplink.js]: We remove the iframe here, but we don't do it in the `timeout`, is that on purpose?

**Pull request merged**